src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,276 @@
+    }
+    private _badgeHorizontalPosition: BadgeHorizontalPosition = this._defaultHorizontalPosition;
+
+    /**
+     * Set if the badge overlaps parent content or flows after parent
+     */
+    @Input() badgeOverlap: boolean = false;
+
+    /**
+     * Max value that can be displayed - if string measures no. of characters,
+     * if number checks against exact number not no. of characters
+     */
+    @Input() badgeMaxValue: number;
+
+    /**
+     * Badge size (based on CSS styles)
+     */
+    private _defaultBadgeSize: BadgeSize = 'medium';
+
+    @Input()
+    get badgeSize(): string {
+        return this._badgeSize;
+    }
+    set badgeSize(size: string) {
+        this._badgeSize = (size.toLowerCase() as BadgeSize) || this._defaultBadgeSize;
+    }
+    private _badgeSize: BadgeSize = this._defaultBadgeSize;
+
+    /**
+     * Hide badge from view
+     */
+    @Input() badgeHidden: boolean = false;
+
+    constructor(
+        private readonly _element: ElementRef<HTMLElement>,
+        private readonly _renderer: Renderer2,
+        private readonly _colorService: ColorService,
+        private readonly _contrastService: ContrastService
+    ) { }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        // get display friendly version of text based on max length and type of val;
+        if (changes.badgeContent) {
+            let finalText: string = changes.badgeContent.currentValue || null;
+
+            if (finalText && this.badgeMaxValue) {
+                if (
+                    this._isNumber &&
+                    changes.badgeContent.currentValue > this.badgeMaxValue
+                ) {
+                    finalText = `${this.badgeMaxValue}+`;
+                } else if (
+                    changes.badgeContent.currentValue.length > this.badgeMaxValue
+                ) {
+                    this._truncatedText = true;
+                    finalText = `${changes.badgeContent.currentValue.substr(
+                        0,
+                        this.badgeMaxValue - 1
+                    )}&hellip;`;
+                } else {
+                    this._truncatedText = false;
+                }
+            }
+
+            this._displayText = finalText;
+        }
+
+        // if text is only set, set aria label from it
+        if (changes.badgeContent && !this.badgeAriaLabel) {
+            this.badgeAriaLabel = changes.badgeContent.currentValue;
+        }
+
+        this.updateBadge();
+    }
+
+    ngOnDestroy(): void {
+        if (this._renderer.destroyNode) {
+            this._renderer.destroyNode(this._badgeElement);
+        }
+    }
+
+    ngAfterContentChecked(): void {
+        this.setXPos();
+    }
+
+    private updateBadge(): HTMLElement {
+        const badgeElement: HTMLSpanElement = this._renderer.createElement(
+            'span'
+        );
+        this.clearExisting();
+
+        badgeElement.classList.add(this._className);
+        badgeElement.classList.add('hidden');
```

Comment:
'hidden' is a bootstrap utility class for hidding items, I was hiding it while it is being manipulated so it doesn't 'shift' around the screen on a slow load.

Reply:
We're avoiding using Bootstrap classes inside components at the moment, with the aim of dropping it as a dependency. For this use case, assuming that it is a problem that you have observed, I would suggest applying `display: none` or `visibility: hidden` directly to the element, which avoids the risk that it will be redefined by the consuming application.

Reply:
Ok sure, I should be keeping this library independent of others as you pointed out. I've changed it to manipulate the display style instead.